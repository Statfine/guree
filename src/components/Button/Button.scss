@import "../../common/utils/index.scss";

.#{scopedClass(button)} {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid currentColor;
  border-radius: 0.3em;
  cursor: pointer;

  & + & {
    margin-left: 8px;
  }
  &.large {
    padding: 10px 30px;
  }
  &.middle {
    padding: 6px 20px;
  }
  &.small {
    padding: 2px 10px;
  }
  > .action + span {
    margin-left: 6px;
  }
  &.default {
    color: #666;
    background-color: #fff;
  }
  &.primary {
    color: #fff;
    background-color: #409eff;
  }
  &.danger {
    color: #fff;
    background-color: #f56c6c;
  }
  &.warning {
    color: #fff;
    background-color: #e6a23c;
  }
  &.text {
    color: #fff;
    background-color: #67c23a;
  }
  &[disabled] {
    opacity: 0.35;
    cursor: default;
  }

  &:focus{
    outline: none;
    opacity: 0.8;
  }
}
