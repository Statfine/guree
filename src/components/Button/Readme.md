### 基本用法

```jsx
<Button>button</Button>
```

### 按钮大小

```jsx
<Button size="large">large</Button>
<Button size="middle">middle</Button>
<Button size="small">small</Button>
```

### 设置图标

```jsx
<Button icon="adn">button</Button>
```

### 加载状态

```jsx
<Button loading>loading</Button>
```

### 禁用状态

```jsx
<Button disabled>disabled</Button>
```

### 按钮类型

```jsx
<Button type="default">default</Button>
<Button type="primary">primary</Button>
<Button type="danger">danger</Button>
<Button type="warning">warning</Button>
<Button type="text">text</Button>
```
