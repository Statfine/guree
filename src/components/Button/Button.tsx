import React from "react";

import "./Button.scss";

/**
 * The only true button.
 */

export interface Props {
  children: string;
}

export default function Button({ children }: Props) {
  return <button className="button">{children}</button>;
}
