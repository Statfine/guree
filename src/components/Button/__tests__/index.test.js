import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../Button'

it('renders correctly', () => {
  const button = renderer
    .create(<Button>button</Button>)
    .toJSON();
  expect(button).toMatchSnapshot();
});
