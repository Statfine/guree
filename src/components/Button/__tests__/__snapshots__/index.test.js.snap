// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 组件 正确导出 Button 组件 1`] = `
<button
  className="guree-button middle default"
>
   
  <span>
    button
  </span>
</button>
`;

exports[`Button 组件 测试 Button 组件 disabled 参数 1`] = `
<button
  className="guree-button middle default"
  disabled={true}
>
   
  <span>
    disabled
  </span>
</button>
`;

exports[`Button 组件 测试 Button 组件 icon 参数 1`] = `
<button
  className="guree-button middle default"
>
  <span
    className="action"
  >
    <svg
      className="guree-icon"
      style={
        Object {
          "fill": undefined,
          "height": "1em",
          "width": "1em",
        }
      }
    >
      <use
        xlinkHref="#adn"
      />
    </svg>
  </span>
   
  <span>
    button
  </span>
</button>
`;

exports[`Button 组件 测试 Button 组件 loading 参数 1`] = `
<button
  className="guree-button middle default"
>
  <span
    className="action"
  >
    <svg
      className="guree-icon"
      style={
        Object {
          "fill": undefined,
          "height": "1em",
          "width": "1em",
        }
      }
    >
      <use
        xlinkHref="#spinner"
      />
    </svg>
  </span>
   
  <span>
    loading
  </span>
</button>
`;

exports[`Button 组件 测试 Button 组件 size 参数 1`] = `
<button
  className="guree-button large default"
>
   
  <span>
    large
  </span>
</button>
`;

exports[`Button 组件 测试 Button 组件 size 参数 2`] = `
<button
  className="guree-button middle default"
>
   
  <span>
    middle
  </span>
</button>
`;

exports[`Button 组件 测试 Button 组件 size 参数 3`] = `
<button
  className="guree-button small default"
>
   
  <span>
    small
  </span>
</button>
`;

exports[`Button 组件 测试 Button 组件 type 参数 1`] = `
<button
  className="guree-button middle default"
>
   
  <span>
    default
  </span>
</button>
`;

exports[`Button 组件 测试 Button 组件 type 参数 2`] = `
<button
  className="guree-button middle primary"
>
   
  <span>
    primary
  </span>
</button>
`;

exports[`Button 组件 测试 Button 组件 type 参数 3`] = `
<button
  className="guree-button middle danger"
>
   
  <span>
    danger
  </span>
</button>
`;

exports[`Button 组件 测试 Button 组件 type 参数 4`] = `
<button
  className="guree-button middle warning"
>
   
  <span>
    warning
  </span>
</button>
`;

exports[`Button 组件 测试 Button 组件 type 参数 5`] = `
<button
  className="guree-button middle text"
>
   
  <span>
    text
  </span>
</button>
`;
